maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java

```diff
@@ -346,4 +342,22 @@ public void testReuse() {
 
         assertEquals(c1, c2, "reused instance should be equivalent to new instance");
     }
+
+    /**
+     * Test <a href="https://issues.apache.org/jira/browse/MNG-7559">MNG-7559</a> edge cases
+     * 1.0.0.RC1 < 1.0.0-RC2
+     * -pfd < final, ga, release
+     * 2.0.1.MR < 2.0.1
+     * 9.4.1.jre16 > 9.4.1.jre16-preview
+     */
+    @Test
+    public void testMng7559() {
+        checkVersionsOrder("1.0.0.RC1", "1.0.0-RC2");
+        checkVersionsOrder("4.0.0.Beta3", "4.0.0-RC2");
+        checkVersionsOrder("2.3-pfd", "2.3");
```

Comment:
`pfd` should be defined why it is *before* `GA`.

Reply:
PFD stands for proposed final draft, but like all qualifiers in semver there are before/less than GA/Release